posthog/hogql_queries/ai/suggested_questions_query_runner.py

```diff
@@ -0,0 +1,92 @@
+from datetime import UTC, timedelta
+
+
+class SuggestedQuestionsQueryRunner(QueryRunner):
+    query: SuggestedQuestionsQuery
+    response: SuggestedQuestionsQueryResponse
+    cached_response: CachedSuggestedQuestionsQueryResponse
+
+    def calculate(self):
+        team = self.team
+        assert team.project is not None
+
+        team_taxonomy_response = TeamTaxonomyQueryRunner(
+            TeamTaxonomyQuery(),
+            team=team,
+            timings=self.timings,
+            modifiers=self.modifiers,
+            limit_context=self.limit_context,
+            query_id=self.query_id,
+        ).calculate()
+
+        messages = [
+            {
+                "role": "system",
+                "content": (
+                    f"You are a product manager at organization {team.organization.name}, handling project {team.project.name}. "
+                    f"This project was created {(timezone.now() - team.project.created_at).total_seconds() // 86400} days ago. "
+                    "Your task is helping product teams understand their users. "
+                    "You guide engineers so that they can make good product decisions themselves."
+                ),
+            },
+            {
+                "role": "user",
+                "content": (
+                    "Here's a list of event types seen in the last 30 days, most popular ones first:\n"
+                    + "\n".join(f"- {e.event} ({e.count} occurrences)" for e in team_taxonomy_response.results)
+                ),
+            },
+            {
+                "role": "user",
+                "content": (
+                    "With this schema in mind, suggest 12 SPECIFIC AND CONCISE QUESTIONS that product teams will find insightful. "
+                    'These questions must be answerable in PostHog. Do not propose placeholders such as "event X", be specific with event names.\n'
+                    'Right now we can only answer questions based on the "events" table. We can use event properties. '
+                    "Note that we can chart trends and create tables. AVOID anything with session duration, event sequences, and correlations.\n"
+                    "Before writing out the question, loosely think out loud through our constraints, possibilities, and needs. "
+                    'Make sure we only propose questions we can answer with our data model. Ignore events prefixed with "$", except $pageview. '
+                    'When done thinking, write "QUESTIONS:", and then the 12 questions, each in its own line, no formatting. '
+                    "Don't number the questions. Questions must be human-friendly but short - you are PENALIZED $10 for every character over 20. "
+                    '(Always abbreviate forms like "what\'s".)'
+                ),
+            },
+        ]
+
+        prompt_tokens_total, completion_tokens_total = 0, 0
+        for _ in range(3):  # Try up to 3 times in case the output is malformed
```

Comment:
FYI. It's a perfect task for structured output. A strict schema will make the error rate very low. Let's leave this as is for now.